File: ruby/test-unit/test/test_simple_example.rb
#--
# Copyright 2016 Testbirds GmbH
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#++

require "selenium/webdriver"
require "test/unit"

# A simple example Selenium test with Test::Unit.
#
# Author:: testbirds
class RubySimpleTest < Test::Unit::TestCase

    # Setup before test.
    def setup
        # use capabilities to specifiy the browser
        capabilities = {
          :javascriptEnabled => TRUE,
          :platform => 'ANY',
          :browserName => 'chrome',
          :version => ''
        }

        # higher timeout to give the VM enough time to get ready 
        client = Selenium::WebDriver::Remote::Http::Default.new
        client.timeout = 300 # in seconds

        # request WebDriver instance from TeCh Selenium to start the VM
        @driver = Selenium::WebDriver.for(:remote,
            :url => "http://localhost:4444/wd/hub",
            :desired_capabilities => capabilities,
            :http_client => client)
    end

    # Test if the title of our demo page is correct.
    def test_title
        @driver.navigate.to "https://demo.testchameleon.com/"
        assert(@driver.title.include? "Gentellela Alela!")
    end

    # Cleanup after test.
    def teardown
        @driver.quit
    end
end

File: python/pytest/test/test_simple_example.py
#!/usr/bin/env python
"""Simple Selenium WebDriver test that uses TestChameleon Selenium"""

import unittest
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

__author__ = "Testbirds GmbH"
__copyright__ = "Copyright 2017 Testbirds GmbH"
__credits__ = ["Testbirds GmbH"]

__license__ = "Apache License"
__version__ = "2.0"
__maintainer__ = "Testbirds GmbH"
__email__ = ""
__status__ = ""

class PythonPytestSimpleTest(unittest.TestCase):
    """A simple Selenium example test with Python and pytest."""

    def setUp(self):
        """Setup before test."""
        capas = DesiredCapabilities.FIREFOX
        self.driver = webdriver.Remote(desired_capabilities=capas)

    def test_title(self):
        """Test the title of our demo page."""
        self.driver.get("https://demo.testchameleon.com/")
        assert "Gentellela Alela!" in self.driver.title

    def tearDown(self):
        """Cleanup after test."""
        self.driver.quit()


if __name__ == "__main__":
    unittest.main()
